app/assets/src/components/views/report/SampleDetailsSidebar/SampleDetailsSidebar.jsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import { keyBy, mapValues } from "lodash";
+import PropTypes from "../../../utils/propTypes";
+import Sidebar from "../../../ui/containers/Sidebar";
+import RemoveIcon from "../../../ui/icons/RemoveIcon";
+import {
+  getSampleMetadata,
+  saveSampleMetadata,
+  getMetadataTypes
+} from "../../../../api";
+import MetadataEditor from "./MetadataEditor";
+import cs from "./sample_details_sidebar.scss";
+
+// Transform the server metadata response to a simple key => value map.
+const processMetadata = metadata => {
+  let newMetadata = keyBy(metadata, "key");
+
+  newMetadata = mapValues(
+    newMetadata,
+    val =>
+      val.data_type === "string"
+        ? val.text_validated_value
+        : val.number_validated_value
+  );
+
+  return newMetadata;
+};
+
+class SampleDetailsSidebar extends React.Component {
+  state = {
+    metadata: null
+  };
+
+  async componentDidMount() {
+    const [metadata, metadataTypes] = await Promise.all([
+      getSampleMetadata(this.props.sample.id),
+      getMetadataTypes()
+    ]);
+    this.setState({
+      metadata: processMetadata(metadata),
+      metadataTypes
+    });
+  }
+
+  onMetadataChange = (key, value, shouldSave) => {
+    this.setState({
+      metadata: {
+        ...this.state.metadata,
+        [key]: value
+      }
+    });
+
+    if (shouldSave) {
+      this._saveHelper(key, value);
```

Comment:
Honestly, as `_saveHelper` signature is almost the same as `saveSampleMetadata` I would just call the later... less confusion.